### PR TITLE
Add methods for classifying and returning the score as well

### PR DESCRIPTION
**Pull request by kreynolds**
I've been using a monkeypatch similar to this on the old classifier for years to help with various kinds of keyword research... hoping to get it (and eventually several others) included upstream with the classifier rebirth

### CODE REVIEW

**Comment by Ch4s3** [comment]
I' like the direction here, it relates to #22, which is awesome. But, I think we should maintain the current method and add `classify_with_score()` as a separate method.

**Comment by kreynolds** [comment]
Ok .. in the interest of reducing code duplication, are on you on board with the idea of having a separate private method that contained the main logic that was then called by both `classify()` and `classify_with_score()`?

I looked at #22 and I think it makes more sense to leave the threshold cutoff in the calling application instead of embedded in `classify()` since threshold failure could be handled differently

**Comment by Ch4s3** [comment]
Yeah I think that's the way to go. What about you @parkr?

**Comment by kreynolds** [comment]
@Ch4s3 should I go ahead and make the change or shall we wait for feedback from @parkr ?

**Comment by parkr** [comment]
Go ahead with the plan.

How would you propose handling threshold failure?

**Comment by kreynolds** [comment]
Handling threshold failure is entirely up to the application .. could be something as simple as (note the scores are completely fabricated, this is just an example):

```ruby
# Return the category for this item, if there is one
category, score = classify_with_score(string)
score > -0.5 ? category : nil

# Raise an exception
category, score = classify_with_score(email_body)
raise SpamError.new("Email marked as spam", score) if category == 'spam' and score > -0.5
```

Basically the threshold handling should be done application side, not library side imo.

**Comment by parkr** [comment]
@kreynolds Thanks for the example. I agree.

**Comment by Ch4s3** [comment]
I like it

**Comment by kreynolds** [comment]
I left the scored_categories method public .. seems like it could be useful in some contexts.

**Comment by kreynolds** [comment]
@Ch4s3 @parkr while I have your attention .. *bump*

File: test/lsi/lsi_test.rb [hunk 1]
```diff
@@ -38,9 +38,34 @@ def test_basic_categorizing
...
+	  assert_in_delta 2.49, lsi.classify_with_score( @str1 )[1], 0.1
+	  assert_in_delta 1.41, lsi.classify_with_score( @str3 )[1], 0.1
+    assert_in_delta 1.99, lsi.classify_with_score( @str5 )[1], 0.1
+	end
+
+  def test_scored_categories
+	  lsi = ClassifierReborn::LSI.new
+	  lsi.add_item @str1, "Dog"
+	  lsi.add_item @str2, "Dog"
+	  lsi.add_item @str3, "Cat"
```
Comment (parkr) [inline test/lsi/lsi_test.rb]: something totally weird with the formatting in this file. could you match your code to whatever the rest of the code is doing?

Comment (kreynolds) [inline test/lsi/lsi_test.rb]: That is weird .. looks the same to me in textmate but obviously there are some actual tabs in there somewhere (whitespace is a bit weird through this whole repo). I'll take a look.

**Pull request merged**